### PR TITLE
Add OASys dates and RoSH summary data to check your answers

### PR DESCRIPTION
**Pull request by patrickjfl**
# Context

[Trello ticket ](https://trello.com/c/517sYjty/631-cya-sa-add-oasys-dates)

# Changes in this PR

This PR renders OASys started, completed and imported dates to the user on the check your answers and submitted application pages. It also renders the RoSH summary data on these pages. I have saved this data on the oasys-import pages for RoSH and risk to self, and created custom responses on the RoSH summary and risk to self 'current risk' pages to render this data. This is not ideal and requires the page to handle data unrelated to current risk, but for now it appears the least intrusive solution.

The PR also removes the 'last updated' OASys data across the service, since we have learned that it is based entirely on the date started and date completed data. Rather than extrapolating to a value with questionable accuracy, we have chosen simply to render the date started and date completed to users.

### Before

Screenshot 2024-01-09 at 15 44 13

Screenshot 2024-01-09 at 15 44 25

### After

Screenshot 2024-01-09 at 15 43 04

Screenshot 2024-01-09 at 15 43 19

# Release checklist

As part of our continuous deployment strategy we must ensure that this work is
ready to be released at any point. Before merging to `main` we must first
confirm:

## Pre merge checklist

- [ ] Are any changes required to the e2e tests?
- [ ] If you've added a new route, have you added a new
  `auditEvent`? (see `server/routes/apply.ts` for examples)
- [ ] Are there environment variables or other infrastructure configuration which needs to be included in this release?
- [ ] Are there any data migrations required. Automatic or manual?
- [ ] Does this rely on changes being deployed to the CAS API?

## Post merge checklist

Once we've merged it will be auto-deployed to the dev environment.

[Find the build-and-deploy job in CircleCI](https://app.circleci.com/pipelines/github/ministryofjustice/hmpps-community-accommodation-tier-2-ui).

Should a release fail at any step, you as the author should now lead the work to
fix it as soon as possible.

### CODE REVIEW

File: server/form-pages/apply/risks-and-needs/risk-of-serious-harm/custom-forms/oasysImport.test.ts [hunk 1]
```diff
@@ -75,6 +75,7 @@ describe('OasysImport', () => {
```
Comment (libuk) [inline server/form-pages/apply/risks-and-needs/risk-of-serious-harm/custom-forms/oasysImport.test.ts:75]: Not sure if this commit needs to be fixed up to another commit or needs a message.

File: server/utils/checkYourAnswersUtils.ts [hunk 2]
```diff
@@ -136,21 +136,27 @@ export const summaryListItemForQuestion = (
   pageKey: string,
   questionAndAnswer: Record<string, string>,
 ) => {
+  const nonEditablePages = ['summary']
+  const nonEditableQuestions = ['OASys started', 'OASys completed', 'OASys imported']
```
Comment (libuk) [inline server/utils/checkYourAnswersUtils.ts:140]: Nice, this is a lot clearer than the solution I went for in CAS3 👍🏾

**Comment by patrickjfl** [comment]
[CYA / SA - Add OASys dates](https://trello.com/c/517sYjty/631-cya-sa-add-oasys-dates)

**Review by libuk — APPROVED** [review]
LGTM
